meeteval/wer/wer/time_constrained.py

```diff
@@ -153,20 +159,26 @@ class TimeMarkedTranscript:
     transcript: List[str]
     timings: List[typing.Tuple[float, float]]
 
-    def check(self, msg):
-        _check_timing_annotations(self.timings, msg)
-
     def has_self_overlaps(self):
-        # Timings are sorted by start time, so it is enough to check
-        # that the end time of each segment is not smaller
-        # than the start time of the next segment
         last_end = 0
-        for t in self.timings:
+        for t in sorted(self.timings):
             if last_end > t[0]:
                 return True
             last_end = t[1]
         return False
 
+    def get_self_overlap(self):
+        last_end = 0
+        self_overlap = 0
+        total = 0
+        for t in sorted(self.timings):
+            if last_end > t[0]:
+                # TODO: Is this the best way to measure self-overlap?
+                self_overlap += 2 * (last_end - t[0])
```

Comment:
Could you add a small test for this function?
I would count the self-overlap only once.

There may be two bugs, could you check:
 -  What happens, when you have two back channels? Here, an example activity pattern:
```
#########################
   ######       
                               ######
 -----------------------> time axis
```
 - What happens for triple overlap?
```
###############
##########
      ###########
```

Reply:
I opted to count the overlap twice because it seems wrong to have timestamps `(0, 2)` and `(0, 2)` but 50% overlap was reported. The intuition could be: Count for how many samples at least one other sample is active at the same time. But as the comment says, I'm not sure whether this is the best way to do it.

Triple overlap is currently not handled correctly, I'll fix that

Reply:
Analternative: Count overlap only once in the total time (which is then smaller than the sum of the segment times) and count overlap once

Reply:
I find it more natural, when self_overlap would be 2 seconds and the total time would be 2 seconds for your example  `(0, 2) and (0, 2)`. In that way, the overlap cannot exceed the duration of the recording with only double overlap.
The implementation could be: Count the overlap without the factor 2 and substract the overlap at the end from the total time. This would automatically handle triple overlap.

Reply:
Done